lib/linguist/languages.yml

```diff
@@ -4856,13 +4856,20 @@ Volt:
   codemirror_mime_type: text/x-d
   language_id: 390
 Vue:
-  type: markup
+  type: programming
   color: "#2c3e50"
   extensions:
   - ".vue"
-  tm_scope: text.html.vue
+  tm_scope: source.vue
   ace_mode: html
+  codemirror_mode: vue
   language_id: 391
+Vue Template:
+  type: markup
```

Comment:
As this is essentially HTML, it's probably a good idea to group it with HTML(add `group: HTML` to this section) like the many other HTML templating languages do.